doc/rst/tools/chpldoc/chpldoc.rst

```diff
@@ -164,6 +164,42 @@ types. A link to the type will be created, if possible. For example:
 For more information see the `Documenting argument, return, and yield values
 and types <#documenting-argument-return-and-yield-values-and-types>`_ section.
 
+Documenting functions that throw
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Errors that are thrown from functions can be documented using the ``:throw:`` or
+``:throws:`` field.  The field will include the ``Error`` subtype being thrown,
```

Comment:
Since the compiler doesn't check that we have this right, it seems optimistic to expect the documentation to be correct with these subtypes being throws... Nonetheless I think it's useful to have a place to describe why an error occurs. 

So:
 * Can the technote explicitly note that these are non-exhaustive and might be more f'instance than anything else?
 * Can the technote show what one should do if they don't want to say what error type is thrown?

Reply:
> Can the technote explicitly note that these are non-exhaustive and might be more f'instance than anything else?

👍 to this. I think the manually curated list of `throws` documentation is always going to be an approximation of the errors that can be thrown, and as a result, we should limit the scope of our documentation effort to documenting only "important" errors and conditions that will throw them, e.g. the errors that are thrown from the function itself.

In the long-term, hopefully we can have an exhaustive list produced by the compiler during the `make docs` process.

Reply:
> Can the technote explicitly note that these are non-exhaustive and might be more f'instance than anything else?

Sure!  I view this document as more for people writing documentation than for people reading the generated documentation, but we can put more explicit recommendations.  Something along the lines of:

> chpldoc will not validate the list of Errors documented for a function.  Documentation writers are responsible for keeping their documentation in line with the behavior of their function.
>
> When writing ``:throws:`` fields, developers do not need to list every possible error that could be thrown - however, specificity will allow users to appropriately anticipate and respond to exceptional situations.  Their response to a PermissionError when accessing a remote file will be different than their response to a ConnectionResetError, for instance.  Documenting such errors ahead of time will allow users to write appropriate ``try``-``catch`` branches.
>
> In some cases, it may be unreasonable to list all the possible Error subclasses that could be thrown.  Developers should feel free to only list a common superclass instead - for instance, ``:throws SystemError:`` as opposed to ``ECONNABORTED``, ``ECONNREFUSED``, and ``ECONNRESET``.